### PR TITLE
Added support for exporting bboxes in VIA-tracks (issue #495) 

### PR DESCRIPTION
**Pull request by harsh-bhanushali-05**
Before submitting a pull request (PR), please read the [contributing guide](https://github.com/neuroinformatics-unit/.github/blob/main/CONTRIBUTING.md).

Please fill out as much of this template as you can, but if you have any problems or questions, just leave a comment and we will help out :)

## Description
This PR introduces support for exporting pose tracking data to the VIA-tracks format, addressing feature request #495. The implementation includes:

 1. A new to_via_tracks_file function for exporting bounding boxes and keypoints in VIA-tracks JSON format.
 2.Support for single-frame and multi-frame inputs, with automatic format conversion (e.g., xywh → xyxy).
 3. Custom video metadata handling with sensible defaults.
 4. Comprehensive unit tests covering functionality, edge cases, and performance.

**What is this PR**
- [ ] Bug fix
- [x] Addition of a new feature
- [ ] Other

**Why is this PR needed?**
This is a feature addition mentioned in the issue #495 
**What does this PR do?**

## References

Please reference any existing issues/PRs that relate to this PR.

## How has this PR been tested?
Wrote testcases to check all the edge cases which could occur 

Please explain how any new code has been tested, and how you have ensured that no existing functionality has changed.
This is a feature addition and no existing code has been changed 
## Is this a breaking change?
No
If this PR breaks any existing functionality, please explain how and why.

## Does this PR require an update to the documentation?
I believe there is already documentation on this. 
If any features have changed, or have been added. Please explain how the
documentation has been updated.
NA

PS. If I have missed something or if something is off I would love to hear reviews and would definitely try to correct the implementation. Cheers! Thanks! 

## Checklist:

- [x] The code has been tested locally
- [x] Tests have been added to cover all new functionality
- [ ] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

**Comment by sfmig** [comment]
hi @harsh-bhanushali-05, thanks for having a go at this!

A few of the CI checks seem to be failing - would you mind investigating these?

**Comment by harsh-bhanushali-05** [comment]
Hey @sfmig! 
I fixed the code and it now passes all the CI testcases. Please share your feedback/any other improvements that i should make.

File: movement/io/save_boxes.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+from pathlib import Path
+
+import numpy as np
+
+# Configure logger
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class Bboxes:
```
[the remark this answers is not in the document]
Comment (harsh-bhanushali-05) [reply movement/io/save_boxes.py]: Hello @sfmig,
I have implemented the suggested updates and believe to have addressed the core requirements. If there are any additional refinements or changes that need attention, I'm eager to tackle them. Thank you for your guidance and support!

**Comment by harsh-bhanushali-05** [comment]
Hey @sfmig,
I noticed that we are shifting to loguru for logging, so i have updated the code in this PR to use loguru instead.

**Comment by niksirbi** [comment]
> Hey @sfmig, I noticed that we are shifting to loguru for logging, so i have updated the code in this PR to use loguru instead.

It's good that you are keeping up-to-date with other PRs @harsh-bhanushali-05.
Bare in mind that though we are swtiching to `loguru`, we are not using it's logger directly as is. The contributing guide has been updated to reflect that, but it's not yet live on our website. Here is the relevant info for you to know:

> We use the `loguru`-based `MovementLogger` for logging. The logger is configured to write logs to a **rotating log file** at the `DEBUG` level and to `sys.stderr` at the `WARNING` level.
> 
> ### Importing the logger
> 
> ```python
> from movement.utils.logging import logger
> ```
> 
> Once the logger is imported, you can log messages with the appropriate severity levels using the same syntax as `loguru`. For example:
> 
> ```python
> logger.debug("Debug message")
> logger.warning("Warning message")
> ```
> 
> ### Logging and raising exceptions
> 
> Both `logger.error()` and `logger.exception()` can be used to log exceptions. The key difference is that `logger.exception()` includes the traceback in the log message.
> 
> Because both methods return the logged exception, you can log and raise the exception in a single line:
> 
> ```python
> raise logger.error(ValueError("message"))
> raise logger.exception(ValueError("message"))  # includes traceback
> ```
> 
> ### When to use `print`, `warnings.warn`, and `logger.warning`
> 
> We aim to adhere to the [[When to use logging](https://docs.python.org/3/howto/logging.html)](https://docs.python.org/3/howto/logging.html) guide to ensure consistency in our logging practices. In general:
> 
> - Use `print` for simple, non-critical messages that do not need to be logged.
> - Use `warnings.warn` for user input issues that are non-critical and can be handled internally by `movement`, such as:
>   - Deprecated function calls that are automatically redirected.
>   - Invalid `fps` values in `ValidPosesDataset` that are implicitly set to `None`.
>   - Processing data with excessive `NaNs` that can be addressed using appropriate methods like `interpolate_over_time()`.
> 
> - Use `logger.warning()` for non-critical issues where default values are being assigned to optional parameters, such as:
>   - `individual_names` or `keypoint_names` in `ValidPosesDataset`.

**Comment by harsh-bhanushali-05** [comment]
Thank you for providing the reference to the contributor's guide.
I changed the implementation of logging statements to use movement.utils.logging instead of loguru directly.
Are any more changes required?

**Comment by sfmig** [comment]
Hi @harsh-bhanushali-05!

This PR is a good attempt at the problem, I liked your modular approach! 

But there is still quite a bit of work to get it to completion. Specifically the test suite should be a bit more aligned with the rest of the codebase. 

Since I am more familiar with this part of the code, I suggest I take it from here and wrap up the pending bits. Note that your contribution will still be recorded and you will be [automatically added to the contributors list ](https://movement.neuroinformatics.dev/community/contributing.html#updating-the-contributors-list). 

I will close this PR now for simplicity, and continue the work [on this PR](https://github.com/neuroinformatics-unit/movement/pull/580). If you are still interested in contributing to movement, feel free to have a look at other open issues - often there are small ones around that may be more approachable!
